### ISSUE TITLE
False positive of E712 in chained comparison

### ISSUE THREAD

**Issue by Joeccp** [PyCQA/pycodestyle#1297]
While implementing my NOR gate:

```python
def NOR(A: bool, B: bool) -> bool:
    return A == B == False

```

pycodestyle (2.14.0) complained:
```text
.\test.py:2:19: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
```

I believe this is a false positive, as the `== False` cannot be replaced with `not (...)` in chained comparison. (I.e., `return not (A == B)` would not work.)

[PEP8](https://peps.python.org/pep-0008/#programming-recommendations) says:
> Don’t compare boolean values to True or False using `==`:
> ```python
> # Correct
> if greeting:
> ```
> ```python
> # Wrong:
> if greeting == True:
> ```
> Worse:
> ```python
> if greeting is True:
> ```

The intention was to avoid unnecessary `== True`/`== False`. However, that is not the case in chained comparison: pycodestyle seems to mistakenly treat `X == Y == Z` as `(X == Y) == Z`.

**Comment by asottile** [comment]
pretty sure you want `a is b is False` and the lint is correct
